src/main/java/mate/academy/service/MovieServiceImpl.java

```diff
@@ -0,0 +1,24 @@
+package mate.academy.service;
+
+import mate.academy.dao.interfaces.MovieDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.Movie;
+
+@Service
+public class MovieServiceImpl implements MovieService {
+    @Inject
+    private MovieDao movieDao;
+
+    @Override
+    public Movie add(Movie movie) {
+        return movieDao.add(movie);
+    }
+
+    @Override
+    public Movie get(Long id) {
+        return movieDao.get(id).orElseThrow(() -> new DataProcessingException("Couldn't find "
+                + "movie by id: " + id));
```

Comment:
```suggestion
        return movieDao.get(id).get();
```

Reply:
In my opinion we should use orElseThrow () method, because theoretically we can get a empty optional